### PR TITLE
Add Simple Task Provider Documentation Guide

### PR DESCRIPTION
**Pull request by pancawardana**
## Summary
Add Simple Task Provider Documentation Guide.

### CODE REVIEW

**Comment by frandi** [comment]
The bottom line of simple task provider that we will create is that it should be "simple". The goal is to show the minimum requirements to create a working task provider without worrying about any optional things.

So let's revise the guide as per my note below:
- please rename the task provider into something that is more descriptive, e.g. `GreetingConsole`. Simply naming it with `SimpleTaskProviders` won't give a clear understanding of what it actually is
- please follow the namespace convention for our task providers which is `Polyrific.Catapult.TaskProviders.*`
- please keep the structure and steps into the bare minimum. e.g. do not include additional configs, do not use lang version 7.1
- remove dependency to PowerShell. Just add a note that we will use a shell to run the `dotnet` command which could be varied depending on the OS. Let's just use `cmd.exe` for Windows.
- remove the steps to create `.vscode` file, just run the app by running `dotnet run` from a command line
- please add a closing statement that the task provider is ready to be installed into OpenCatapult, and give a link to the guide page

Thank you.

**Comment by pancawardana** [comment]
> The bottom line of simple task provider that we will create is that it should be "simple". The goal is to show the minimum requirements to create a working task provider without worrying about any optional things.
> 
> So let's revise the guide as per my note below:
> 
> * please rename the task provider into something that is more descriptive, e.g. `GreetingConsole`. Simply naming it with `SimpleTaskProviders` won't give a clear understanding of what it actually is
> * please follow the namespace convention for our task providers which is `Polyrific.Catapult.TaskProviders.*`
> * please keep the structure and steps into the bare minimum. e.g. do not include additional configs, do not use lang version 7.1
> * remove dependency to PowerShell. Just add a note that we will use a shell to run the `dotnet` command which could be varied depending on the OS. Let's just use `cmd.exe` for Windows.
> * remove the steps to create `.vscode` file, just run the app by running `dotnet run` from a command line
> * please add a closing statement that the task provider is ready to be installed into OpenCatapult, and give a link to the guide page
> 
> Thank you.

**Comment by pancawardana** [comment]
Well noted @frandi, I will update it accordingly.

**Comment by pancawardana** [comment]
Updated, please have a look @frandi @BrainBalaka.

File: docs/dev-guides/simple-task-provider.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+```csharp
+static void Main(string[] args)
+{
+    var app = new Program(args);
+    var result = app.Execute().Result;
+    Console.Write(result);
+}
+```
+
+The next thing is to write the logic of our code generator inside the `Generate` method. Here, we will only extract the configuration  and additional configurations that are passed into the task provider. We'd also need to get the `Config` property in the base class, to determine where our source code will be saved. Our user can enter their prefered location in `Config.OutputLocation` but if they have no preferences, we'd take the `Config.WorkingLocation` instead.
```
Comment (BrainBalaka) [inline docs/dev-guides/simple-task-provider.md]: Remove the part about Additional configuration in this paragraph, as well as the one in the code

File: docs/dev-guides/simple-task-provider.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+## Prerequisites
+- A code editor. We will use [Visual Studio Code](https://code.visualstudio.com/download) in this example
+- [dotnet core sdk version 2.1](https://dotnet.microsoft.com/download/dotnet-core/2.1)
+- If you are on a non-windows OS, you'd need to install powershell [here](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell?view=powershell-6#powershell-core)
+
+## Create the provider project
+Create a new dotnet core console project by using the dotnet cli:
+```sh
+dotnet new console --name Polyrific.Catapult.SimpleTaskProviders
```
[the remark this answers is not in the document]
Comment (BrainBalaka) [reply docs/dev-guides/simple-task-provider.md]: @pancawardana I think the name should be: `Polyrific.Catapult.TaskProviders.SimpleGenerator`

**Comment by pancawardana** [comment]
Hi @BrainBalaka, updated. Please have a look.

**Pull request merged**